app/main.go

```diff
@@ -38,13 +68,27 @@ func main() {
 	mux := http.NewServeMux()
 
 	fileServer := http.FileServer(http.Dir(webDir + "/static/"))
+	awsBaseUrl := env.Get("AWS_BASE_URL", "http://localstack:4566")
+
+	secretsClient := &secrets.Client{BaseURL: awsBaseUrl}
+
+	signInClient := signin.NewClient(http.DefaultClient, "/auth/callback", secretsClient)
+	err = signInClient.Discover(issuer.String() + "/.well-known/openid-configuration")
+	if err != nil {
+		logger.Fatal(err)
+	}
 
 	mux.Handle("/static/", http.StripPrefix("/static", fileServer))
+
+	mux.Handle("/login", page.Login(*signInClient, appPublicURL, clientID, signInPublicURL))
+	mux.Handle("/home", page.Home(tmpls.Get("home.gohtml"), fmt.Sprintf("%s/login", appPublicURL), bundle.For("en"), page.En))
+	mux.Handle("/auth/callback", page.SetToken(*signInClient, appPublicURL, clientID, RandomString(12)))
```

Comment:
we'll want that value to be different each time, so could pass the func in and let `SetToken` call it

Reply:
I've moved this to its own package as we'll probably get some re-use out of it (or at least have the place to add different types of random funcs in) with `state` and `nonce`:

<img width="242" alt="Screenshot 2022-08-25 at 11 52 42" src="https://user-images.githubusercontent.com/17926619/186646469-8b0da537-9332-432e-b118-26da3bdd6f31.png">

So the API is like `random.String(12)`. Any thoughts on this? I know utility-type classes can sometimes divide the room.

Reply:
I guess that works, although you could also move it to `signin` since that is the only thing using it at the moment